front/components/data_source/CreateConnectionSnowflakeModal.tsx

```diff
@@ -0,0 +1,194 @@
+                icon={BookOpenIcon}
+              />
+            </a>
+            {connectorProviderConfiguration.limitations && (
+              <div className="flex flex-col gap-y-2">
+                <div className="grow text-sm font-medium text-element-800">
+                  Limitations
+                </div>
+                <div className="text-sm font-normal text-element-700">
+                  {connectorProviderConfiguration.limitations}
+                </div>
+              </div>
+            )}
+
+            <Page.SectionHeader title="Snowflake Credentials" />
+
+            {error && (
+              <div className="w-full rounded-md bg-red-100 p-4 text-red-800">
+                {error}
+              </div>
+            )}
+
+            <div className="w-full space-y-4">
+              <Input
+                label="Snowflake Account identifier"
+                name="username"
+                value={credentials.account}
+                placeholder="au12345.us-east-1"
+                onChange={(value) => {
+                  setCredentials({ ...credentials, account: value });
+                  setError(null);
+                }}
+              />
+              <Input
+                label="Role"
+                name="username"
+                value={credentials.role}
+                placeholder="dev_role"
+                onChange={(value) => {
+                  setCredentials({ ...credentials, role: value });
+                  setError(null);
+                }}
+              />
+              <Input
+                label="Warehouse"
+                name="warehouse"
+                value={credentials.warehouse}
+                placeholder="dev_warehouse"
+                onChange={(value) => {
+                  setCredentials({ ...credentials, warehouse: value });
+                  setError(null);
+                }}
+              />
+              <Input
+                label="Username"
+                name="username"
+                value={credentials.username}
+                placeholder="dev_user"
+                onChange={(value) => {
+                  setCredentials({ ...credentials, username: value });
+                  setError(null);
+                }}
+              />
+              <Input
+                label="Password"
+                name="password"
+                isPassword={true}
+                value={credentials.password}
+                placeholder=""
+                onChange={(value) => {
+                  setCredentials({ ...credentials, password: value });
+                  setError(null);
+                }}
+              />
+            </div>
+
+            <div className="flex justify-center pt-2">
+              <Button.List isWrapping={true}>
+                <Button
+                  variant="primary"
+                  size="md"
+                  icon={CloudArrowLeftRightIcon}
+                  onClick={async () => {
```

Comment:
nit: the async / await here is not doing anything (onClick itself is not awaited)